sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -694,12 +740,7 @@ export class MessageSender extends LinkEntity {
    * @returns {Promise<number>}
    * @memberof MessageSender
    */
-  async getMaxMessageSize(
-    options: {
-      retryOptions?: RetryOptions;
-    } = {}
-  ): Promise<number> {
-    const retryOptions = options.retryOptions || {};
+  async getMaxMessageSize(): Promise<number> {
```

Comment:
What is the criteria for when a method takes it's own OperationOptions vs only inheriting it from the service client?

Reply:
https://github.com/Azure/azure-sdk-for-js/pull/8159#issuecomment-608254608

Reply:
OperationOptions are never inherited from the client. They are specific to each operation/method.

Here, `getMaxMessageSize()` is not a method user would call. Its called from `createBatch()` which in turn is what the user would call. So, `createBatch()` gets to have `OperationOptions`.

I didnt pass it to `getMaxMessageSize()` because all this function does is initialize the link and we dont yet have the way to cancel an initialization that is in progress. See #4422